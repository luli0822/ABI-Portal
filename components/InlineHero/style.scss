@import "../../scss/layout-mixins";
@import "../../scss/text-mixins";
@import "../../scss/button-link-mixins";
@import "../../scss/a11y-mixins";

$root-class: "inline-hero";

.#{$root-class} {
  // max-width-section contains the background to the section max-width. Not necessary if your background is 100% width
  @include max-width-section();
  // standard section spacing - not always necessary, not always the best way to provide this (i.e. you might prefer a variable to use in padding or margin as necessary per design)
  @include section-bottom-spacing();

  // most designs won't use base-font-size for this, but here's an example of how you might dynamically build paddings and margins
  padding-top: $base-font-size * 2px;
  padding-bottom: $base-font-size * 2px;

  &__content {
    // this container() provides the max-width that constrains the content, as well as padding and margin needed for the rows and outside gutter
    @include container();

    position: relative;
    padding: 2 * $grid-gutter-width;
  }

  &__cover-image {
    position: absolute;
    overflow: hidden;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: -1;
    display: flex;
  }

  &__image {
    flex: 0 0 100%;
  }

  &__image-screen {
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    width: 100%;
    height: 100%;
    opacity: 0.5;
    background-color: black;
  }

  &__heading {
    @include heading-two();

    color: $white;
    text-align: center;
  }

  &__detail-row {
    // make-row provides the margin and padding necessary for a row
    @include make-row();
  }

  &__rich-p {
    @include body();

    color: white;
    text-align: center;
  }

  &__rich-b {
    font-weight: bold;
  }

  &__rich-i {
    font-style: italic;
  }

  &__rich-li {
    list-style-type: disc;
  }

  &__cta-container {
    display: flex;
    justify-content: space-around;
  }

  &__cta {
    // includes the mixin to make the __cta element use the global cta-button-primary styles
    @include cta-button-primary();

    display: inline-block;
  }
}
