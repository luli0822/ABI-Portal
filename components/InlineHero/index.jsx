import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';
import ContentfulLink from '../../contentful/ContentfulLink';
import ContentfulImage from '../../contentful/ContentfulImage';
import RichText from '../../contentful/RichText';
import { contentPropType } from '../../contentful/prop-types';

function InlineHero(props) {
	const { content, isFirstChild } = props;
	const rootClass = 'inline-hero';

	// components don't know where they are on the page, but the page must always have an H1
	// this is a simple way to show the H1 using the first component on the page
	// note this doesn't work if there is no title content in the first component
	const HTag = isFirstChild ? 'h1' : 'h2';

	return (
		<section className={rootClass}>
			<div className={`${rootClass}__content`}>
				{content.primaryMedia && (
					<div className={`${rootClass}__cover-image`}>
						<ContentfulImage asset={content.primaryMedia} className={`${rootClass}__image`} cover />
						<div className={`${rootClass}__image-screen`} />
					</div>
				)}
				{content.title && <HTag className={`${rootClass}__heading`}>{content.title}</HTag>}
				{content.body && <RichText document={content.body} rootClass={`${rootClass}__rich`} />}

				<div className={`${rootClass}__cta-container`}>
					{content.primaryLink && (
						<ContentfulLink linkEntry={content.primaryLink} className={`${rootClass}__cta`}>
							{content.primaryLinkText}
						</ContentfulLink>
					)}
					{content.secondaryLink && (
						<ContentfulLink linkEntry={content.secondaryLink} className={`${rootClass}__cta`}>
							{content.secondaryLinkText}
						</ContentfulLink>
					)}
				</div>
			</div>
		</section>
	);
}

InlineHero.propTypes = {
	content: contentPropType.isRequired,
	isFirstChild: PropTypes.bool.isRequired,
};

export default InlineHero;
