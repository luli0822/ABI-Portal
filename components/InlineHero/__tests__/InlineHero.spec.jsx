import React from 'react';
import { shallow } from 'enzyme';
import InlineHero from '../index';

describe('<InlineHero />', () => {
	const props = {
		content: {
			primaryMedia: {},
			title: 'some title',
			body: {},
			primaryLink: {},
			primaryLinkText: 'some text',
			secondaryLink: {},
			secondaryLinkText: 'some text',
		},
		isFirstChild: false,
	};
	it('renders correctly', () => {
		// https://github.com/facebook/jest/issues/7802
		// using .debug() instead of adding dependency on enzyme-to-json for now
		// eslint-disable-next-line react/jsx-props-no-spreading
		expect(shallow(<InlineHero {...props} />).debug()).toMatchSnapshot();
	});
});
