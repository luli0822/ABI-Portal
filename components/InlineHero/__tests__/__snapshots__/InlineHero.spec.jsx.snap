// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineHero /> renders correctly 1`] = `
"<section className=\\"inline-hero\\">
  <div className=\\"inline-hero__content\\">
    <div className=\\"inline-hero__cover-image\\">
      <ContentfulImage asset={{...}} className=\\"inline-hero__image\\" cover={true} alt={[undefined]} lazyload={true} />
      <div className=\\"inline-hero__image-screen\\" />
    </div>
    <h2 className=\\"inline-hero__heading\\">
      some title
    </h2>
    <RichText document={{...}} rootClass=\\"inline-hero__rich\\" replacements={{...}} />
    <div className=\\"inline-hero__cta-container\\">
      <ContentfulLink linkEntry={{...}} className=\\"inline-hero__cta\\" targetLang={[undefined]}>
        some text
      </ContentfulLink>
      <ContentfulLink linkEntry={{...}} className=\\"inline-hero__cta\\" targetLang={[undefined]}>
        some text
      </ContentfulLink>
    </div>
  </div>
</section>"
`;
